aya/src/programs/tc.rs

```diff
@@ -118,25 +164,41 @@ impl SchedClassifier {
         load_program(BPF_PROG_TYPE_SCHED_CLS, &mut self.data)
     }
 
-    /// Attaches the program to the given `interface` using the default options.
+    /// Attaches the program to the given `interface` using the TCX link
+    /// API in the first position by default, revert to the legacy netlink API otherwise.
+    /// For fine-grain control over link ordering use [`SchedClassifier::attach_with_options`].
     ///
     /// The returned value can be used to detach, see [SchedClassifier::detach].
     ///
     /// # Errors
     ///
-    /// [`TcError::NetlinkError`] is returned if attaching fails. A common cause
-    /// of failure is not having added the `clsact` qdisc to the given
+    /// [`TcError::NetlinkError`] is returned if attaching via netlink fails. A
```

Comment:
What happens when attaching with TCX fails?

Reply:
Good catch So there will be a syscall error for kernels >= 6.6.0 added a note